modules/react/common/stories/BasicCompoundComponent.stories.mdx

```diff
@@ -0,0 +1,286 @@
+yarn add @workday/canvas-kit-react
+```
+
+# createModelHook
+
+When building compound components that might have some internal state or events, you typically want
+a model. A model allows a component to share information. This is where `createModelHook` comes in
+handy.
+
+## Usage
+
+`createModelHook` is a factory function that takes in `defaultConfig` and `requiredConfig` and
+returns a function which is your model hook.
+
+If we take diclosure as an example, this is how we'd use `composeModelHook`
+
+```tsx
+import {createModelHook} from '@workday/canvas-kit-react/common';
+export const useDisclosureModel = createModelHook({
+  // This becomes the default values on the model
+  defaultConfig: {
+    // some default config
+  },
+})(config => {
+  const state = {
+    // some state
+  };
+  // Sets events that can be used across subcomponents
+  const events = {
+    // some events
+  };
+
+  return {state, events};
+});
+```
+
+<ExampleCodeBlock code={CreateModelHook} />
+
+Typescript will infer all config from the returned `state`, `events`, `defaultConfig` and the
+`requiredConfig`.
+
+## API
+
+When `useDisclosureModel`is created, 5 properties are attached to it for composability:
+`defaultConfig`, `requiredConfig`, `TConfig`, `getElemProps`, `Context`.
+
+### defaultConfig
+
+All config that has default values. Optional values also go here, but are represented as undefined
+as part of the union type `(undefined as undefined | string)`. Using this can be useful when
+creating other models that share similar config for example:
+
+```tsx
+import {createModelHook} from '@workday/canvas-kit-react/common';
+export const useExpandableModel = createModelHook({
+  defaultConfig: {
+    // extend the default config from the `useDisclosureModel`
+    ...useDisclosureModel.defaultConfig,
+  },
+})(config => {
+  const disclosure = useDisclosureModel(config);
+  const state = {
+    ...disclosure.state,
+  };
+  const events = {
+    ...disclosure.events,
+  };
+  return {state, events};
+});
+```
+
+### requiredConfig
+
+All config that is required by the model. Most config should go into default config with thoughtful
+defaults
+
+### contextOverride
+
+The React context used by components to share a model between a container component and
+subcomponents. It is used internally by createContainer and createSubcomponents and should not need
+to be referenced directly.
+
+Each subcomponent is tied to a model hook's context. For example, useModalModel extends
+usePopupModel, but <Modal> components point directly to <Popup> components. The contextOverride:
+usePopupModel.Context forces useModalModel's internal context to point to the same context reference
+as the one used in usePopupModel. This allows the model returned by useModalModel to be compatible
+with both <Modal> subcomponents and <Popup> subcomponents. If you do not override context, you must
+create a new container and subcomponent for every component using the newly created model hook.
```

Comment:
Much of this suggestion is adding backticks around function names, though I also had to remove the angled brackets around the component names to get the MDX to compile properly (otherwise it was treating them as actual JSX components). This was preventing the examples from rendering at all in the Storybook nav.
```suggestion
The React context used by components to share a model between a container component and
subcomponents. It is used internally by `createContainer` and `createSubcomponents` and should not
need to be referenced directly.

Each subcomponent is tied to a model hook's context. For example, `useModalModel` extends
`usePopupModel`, but `Modal` components point directly to `Popup` components. The
`contextOverride: usePopupModel.Context` forces `useModalModel`'s internal context to point to the
same context reference as the one used in `usePopupModel`. This allows the model returned by
`useModalModel` to be compatible with both `Modal` subcomponents and `Popup` subcomponents. If you
do not override context, you must create a new container and subcomponent for every component using
the newly created model hook.
```